EIPS/eip-7685.md

```diff
@@ -0,0 +1,76 @@
+---
+eip: 7685
+title: General purpose execution layer requests
+description: A general purpose bus for sharing EL triggered requests with the CL
+author: lightclient (@lightclient)
+discussions-to: https://ethereum-magicians.org/t/eip-7685-general-purpose-execution-layer-requests/19668
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-04-14
+---
+
+## Abstract
+
+This proposal defines a general purpose framework for propagating contract-triggered requests from execution layer to the consensus layer. It extends the header and body with only a single field each to store the request information.
+
+## Motivation
+
+The proliferation of smart contract controlled validators has caused there to be a demand for additional EL triggered behaviors. By allowing these systems to delegate administrative operations to their governing smart contracts, they can avoid intermediaries needing to step in and ensure certain operations occur. This creates a safer system for end users.
+
+## Specification
+
+### Execution Layer
+
+#### Request
+
+A `Request` consists of a `RequestType` prepended to an opaque byte array `RequestData`.
+
+```
+Request = RequestType ++ RequestData
+```
+
+Let `Requests` be the list of all `Request` objects in the block in ascending order by type.
+
+#### Header
+
+Extend the header with a new 32 byte value `RequestsRoot`.
+
+Let `RequestsRoot` be the root of a Merkle-Patricia trie keyed by the index in the list of `Requests`. This is equivalent to how the transaction trie root is computed.
+
+### Consensus Layer
+
+Each proposal may choose how to extend `ExecutionPayload` to include the new EL request.
+
+A additional processing step is be added to `process_execution_payload` to iterate over and process all requests.
```

Comment:
some things we will want to process before `process_execution_payload` and some things we will want to process after

iirc none of the current 'requests' are actually addressed _inside_ `process_execution_payload`

given the validity semantics of the spec, this isn't really material to the structure introduced by this EIP

Reply:
okay i will remove this

Reply:
#8451